miniupnpc/miniwget.c

```diff
@@ -230,102 +230,69 @@ getHTTPResponse(int s, int * size, int * status_code)
 			memcpy(buf, header_buf + endofheaders, n);
 			/* if(headers) */
 		}
-		if(endofheaders)
+//		if(endofheaders) always true here
```

Comment:
If you look at conditional operator starting at line 116, all assignments to _endofheaders_ are performed above the line 159, where it is checked again and in case of 0 _continue_ would be executed.